extras/kms/kms.go

```diff
@@ -241,69 +244,66 @@ func (k *Kms) GetWrapper(ctx context.Context, scopeId string, purpose KeyPurpose
 	if err != nil {
 		return nil, fmt.Errorf("%s: error loading %q for scope %q: %w", op, purpose, scopeId, err)
 	}
-	if err := k.addKey(ctx, scopeWrapperCache, purpose, wrapper, WithKeyId(scopeId+string(purpose))); err != nil {
+	if err := k.addKey(ctx, scopeWrapperCache, purpose, wrapper, WithKeyVersionId(scopeId+string(purpose))); err != nil {
 		return nil, fmt.Errorf("%s: error adding key to cache: %w", op, err)
 	}
 
-	if opts.withKeyId != "" {
-		if keyIdWrapper := wrapper.WrapperForKeyId(opts.withKeyId); keyIdWrapper != nil {
-			return keyIdWrapper, nil
+	if opts.withKeyVersionId != "" {
+		if keyIdVersionWrapper := wrapper.WrapperForKeyId(opts.withKeyVersionId); keyIdVersionWrapper != nil {
+			return keyIdVersionWrapper, nil
 		}
-		return nil, fmt.Errorf("%s: unable to find specified key ID: %w", op, ErrKeyNotFound)
+		return nil, fmt.Errorf("%s: unable to find specified key version ID: %w", op, ErrKeyNotFound)
 	}
 
 	return wrapper, nil
 }
 
-// ListKeys returns the current list of kms keys
+// ListKeys returns the current list of kms keys.
 func (k *Kms) ListKeys(ctx context.Context, scopeId string) ([]Key, error) {
```

Comment:
I reverted this back to the old signature, which makes it less of a breaking now. Now the only break I believe is the removal of the `Version` field from the `Key` type.